data/basics/teal_test.go

```diff
@@ -221,4 +223,29 @@ func TestEvalDeltaEqual(t *testing.T) {
 		},
 	}
 	a.False(d1.Equal(d2))
+
+	d2 = EvalDelta{
+		Logs: []LogItem{{ID: 0, Message: "val"}},
+	}
+	a.False(d1.Equal(d2))
+
+	d1 = EvalDelta{
+		Logs: []LogItem{{ID: 0, Message: "val2"}},
+	}
+	a.False(d1.Equal(d2))
+
+	d1 = EvalDelta{
+		Logs: []LogItem{{ID: 1, Message: "val"}},
+	}
+	a.False(d1.Equal(d2))
+
+	d1 = EvalDelta{
+		Logs: []LogItem{{ID: 1, Message: "val2"}},
+	}
+	a.False(d1.Equal(d2))
+
+	d1 = EvalDelta{
+		Logs: []LogItem{{ID: 0, Message: "val"}},
+	}
+	a.True(d1.Equal(d2))
```

Comment:
Suggested test case: compare two `EvalDelta`s with logs `[A]` and `[A, B]` -- I believe this would currently pass because of the length issue I mentioned